clang/test/dpct/helper_files_ref/include/dpl_extras/dpcpp_extensions.h

```diff
@@ -501,6 +501,25 @@ reduce(Item item, T (&inputs)[VALUES_PER_THREAD], BinaryOperation binary_op) {
   return detail::__reduce_over_group(item.get_group(), result, binary_op);
 }
 
+/// Perform a reduction on a limited number of the work items in a subgroup
+///
+/// \param item A work-item in a group.
+/// \param value value per work item which is to be reduced
+/// \param items_to_reduce num work items at the start of the subgroup to reduce
+/// \param identity identity for type and binary_op
+/// \param binary_op functor that implements the binary operation used to
+/// perform the scan. \returns value of the reduction using binary_op
+template <typename Item, typename T, class BinaryOperation>
+__attribute__((always_inline)) T
+reduce_over_partial_group(const Item &item, const T &value,
```

Comment:
This looks good! In my opinion, if we can't find a correct identifiy for the binop, we can just emit a warning to tell users that they should manually specify an identify. So, I think, we can remove the `sycl::has_known_identity` constraint, WDYT? @danhoeflinger @zhimingwang36

Reply:
@yihanwg The `sycl::reduce_over_group` function itself requires that the binary op / type be from this set.  Even if the user specified an identity, we would not be able to use it.  
To handle other cases, we would need to write our own reduction code and it would require local memory to be passed in.

Reply:
I suppose (as used below in this same file) there is a `sycl::ext::oneapi::experimental::joint_reduce()` we can use which allows for user defined reductions, but we would still need to create local memory and pass it in in that scenario via the grouphelper scratchpad.